cmd/algorand-indexer/daemon.go

```diff
@@ -505,7 +511,7 @@ func blockHandler(proc processor.Processor, retryDelay time.Duration) func(conte
 			case <-ctx.Done():
 				return err
 			case <-time.After(retryDelay):
-				break
+				// NOOP
```

Comment:
@tzaffi Sanity checking:  Is the removal of the `break` statement by-design?

I'm raising the note because at surface level, the change _seems_ unrelated to Boxes + impacts a seemingly central code path.  I'm wondering if perhaps the change was intended for local-only testing.

Reply:
After catching as similar lint problem in an earlier PR, I cavalierly assumed, that the same problem occurs here. I.e., a `break` is used in a c-style. It's clear that the [behavior is the same](https://go.dev/play/p/kkcq4Qvi330). Also, the comment on line 509:
```
// Delay or terminate before next attempt.
```
`Delay` corresponds to the 2nd part of the switch. However, I think we should get others more familiar with the code to have a look as well.

@AlgoStephenAkiki @winder @shiqizng

Reply:
This change is fine, Zeph's suggestion is probably a little more conventional for go programs than having the break.

This code is needed because Indexer takes total control of incoming signals and refuses to return until a graceful shutdown has completed. So when we want a delay between attempts (in case of failure), we need to do so while also checking whether the program needs to terminate. Thus, we check for the `ctx.Done()` signal while waiting for `time.After(retryDelay)` to elapse.